src/FormattingSettingsService.ts

```diff
@@ -159,24 +155,45 @@ export class FormattingSettingsService implements IFormattingSettingsService {
         return formattingModel;
     }
 
-    private buildFormattingSlices({slices, objectName, sliceNames, formattingSlices }: IBuildFormattingSlicesParams) {
+    private buildFormattingSlices({ slices, objectName, formattingSlices }: IBuildFormattingSlicesParams) {
+        // In case formatting model adds data points or top categories (Like when you modify specific visual category color).
+        // these categories use same object name and property name from capabilities and the generated uid will be the same for these formatting categories properties
+        // Solution => Save slice names to modify each slice uid to be unique by adding counter value to the new slice uid
+        const sliceNames: { [name: string]: number } = {};
+
+        // We are also supporting slices with the same name if they differ in the selector field (not the altConstantSelector).
+        // If the slices with the same name have different selectors we keep the uid as is, else we change it.
+        // We recommend using different selectors in the selector field for slices with the same name.
+        const selectorsMap: Record<string, { [selector: string]: boolean }> = {}
+
         // Filter slices based on their visibility
-        slices?.filter(({visible = true}) => visible)
+        slices?.filter(({ visible = true }) => visible)
             .forEach((slice: Slice) => {
-            let formattingSlice: visuals.FormattingSlice = slice?.getFormattingSlice(objectName, this.localizationManager);
-
-            if (formattingSlice) {
-                // Modify formatting slice uid if needed
-                if (sliceNames[slice.name] === undefined) {
-                    sliceNames[slice.name] = 0;
-                } else {
-                    sliceNames[slice.name]++;
-                    formattingSlice.uid = `${formattingSlice.uid}-${sliceNames[slice.name]}`;
-                }
+                let formattingSlice: visuals.FormattingSlice = slice?.getFormattingSlice(objectName, this.localizationManager);
+
+                if (formattingSlice) {
+                    // Modify formatting slice uid if needed
+                    const sliceName = slice.name;
+                    const sliceSelector = (slice as SimpleSlice).selector;
+                    if (sliceNames[slice.name] === undefined) {
+                        selectorsMap[sliceName] = {};
+                        sliceNames[slice.name] = 0;
+                        if (sliceSelector) {
+                            selectorsMap[slice.name][JSON.stringify(sliceSelector)] = true;
```

Comment:
Create a variable for `JSON.stringify(sliceSelector)` as it is used in three places